node/src/components/gossiper/metrics.rs

```diff
@@ -0,0 +1,109 @@
+use prometheus::{IntCounter, IntGauge, Registry};
+
+/// Metrics for the gossiper component.
+#[derive(Debug)]
+pub struct GossiperMetrics {
+    /// Total number of items received by the gossiper.
+    pub(super) items_received: IntCounter,
+    /// Total number of items gossipped onwards after receiving.
+    pub(super) items_regossipped: IntCounter,
+    /// Number of times the process had to pause due to running out of peers.
+    pub(super) times_ran_out_of_peers: IntCounter,
+    /// Number of items in the gossip table that are paused.
+    pub(super) tbl_items_paused: IntGauge,
+    /// Number of items in the gossip table that are currently being gossiped.
+    pub(super) tbl_items_current: IntGauge,
+    /// Number of items in the gossip table that are currently being finished.
+    pub(super) tbl_items_finished: IntGauge,
+
+    /// Reference to the registry for unregistering.
+    registry: Registry,
+}
+
+impl GossiperMetrics {
+    /// Creates a new instance of gossiper metrics, using the given prefix.
+    pub fn new(name: &str, registry: &Registry) -> Result<Self, prometheus::Error> {
+        let items_received = IntCounter::new(
+            format!("{}_items_received", name),
+            format!(
+                "number of items received by the {} gossiper component",
+                name
+            ),
+        )?;
+        let items_regossipped = IntCounter::new(
+            format!("{}_items_regossipped", name),
```

Comment:
Why no just `_gossiped`?

Reply:
"gossipped" (or gossiped? Somehow both spellings seem to show up in online dictionaries) would imply that this also includes items that were sent by us initially. This metric however only includes items that were received and then gossipped onwards.

By subtracting `items_gossipped` from `items_received` you can determine how many items reached their final destination at our node.

Reply:
I didn't mean to pick on the spelling, I was wondering if we can drop the `re_` prefix. It won't clash with anything.

> By subtracting items_gossipped from items_received you can determine how many items reached their final destination at our node.

That's a good point but I would call it a number of new events.

Reply:
We might want to define an `items_gossipped` metric at some point (but we'd have to figure out what it is supposed to mean first :)) That's why I am in favor of keeping the `re`-prefix for now.

Reply:
I don't think there's a distinction between _re-gossip_ vs _gossip_ but I won't argue about it.